src/keboola/facebook/api/request.clj

```diff
@@ -200,6 +201,45 @@
         :body-data [(if (not-empty path) {sanitized-path response-body} response-body)]
         :response (if (not-empty path) {sanitized-path response-body} response-body)}))))
 
+(defn- job-finished? [poll-response]
+  (let [status (-> poll-response :body :async_status)
+        completed? (= status "Job Completed")
+        failed? (some (partial = status)  ["Job Failed" "Job Skipped"])]
+    (cond
+      (not status) (runtime/app-error (str "Polling failed with unknown status" (:body poll-response)))
+      failed?  (runtime/user-error (str "Polling failed with status:" status (:body poll-response)))
+      completed? (do (log-strings "Polling finished with status:" status) (:body poll-response))
+      :else
+      nil)))
+
+(defn- poll-async-request [report_run_id access-token version]
+  (let [url (str (make-url report_run_id version) "?access_token=" access-token)
+        request-fn #(client/GET url :as :json)
+        finished?-fn #(job-finished? (request-fn))]
+    (log-strings "Started polling for insights job report:" report_run_id)
+    (with-exp-backoff finished?-fn)))
+
+(defn- collect-async-insights-result [access-token ad-account-id report-run-id version]
+  (let [url (str (make-url report-run-id version) "/insights/?access_token=" access-token)
+        response (make-get-request url)
+        response-body (:body response)]
+    ;(println "response-body" response-body)
+    (page-and-collect
+     {:ex-account-id ad-account-id
+      :parent-id ad-account-id
+      :fb-graph-node "page"
+      :table-name "page"
+      :path "insights"
+      :body-data [{"insights" response-body}]
+      :response nil})))
```

Comment:
pouzitie fce `page-and-collect` zkopirovane z existujucej implementacie pre "nested query"(sync query)